tests/features/payment-gateway-suggestions/data-source-poller.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Test the data source poller for payment gateway suggestions.
+ *
+ * @package WooCommerce\Admin\Tests\PaymentGatewaySuggestions
+ */
+
+use Automattic\WooCommerce\Admin\Features\PaymentGatewaySuggestions\Init as PaymentGatewaySuggestions;
+use Automattic\WooCommerce\Admin\Features\PaymentGatewaySuggestions\DataSourcePoller;
+
+/**
+ * class WC_Tests_PaymentGatewaySuggestions_DataSourcePoller
+ */
+class WC_Tests_PaymentGatewaySuggestions_DataSourcePoller extends WC_Unit_Test_Case {
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		add_filter(
+			DataSourcePoller::FILTER_NAME,
+			function() {
+				return array(
+					'payment-gateway-suggestions-data-source.json',
+				);
+			}
+		);
+
+		add_filter(
+			'pre_http_request',
+			function( $pre, $parsed_args, $url ) {
+				if ( 'payment-gateway-suggestions-data-source.json' === $url ) {
+					return array(
+						'body' => wp_json_encode(
+							array(
+								array(
+									'id' => 'mock-gateway1',
+								),
+								array(
+									'id' => 'mock-gateway2',
+								),
+								array(
+									'key' => 'mock-gateway-invalid',
+								),
+							),
+						),
```

Comment:
@joshuatf there seems to be a syntax error on this line:
`PHP Parse error:  syntax error, unexpected ')' in /tmp/wordpress/wp-content/plugins/woocommerce-admin/tests/features/payment-gateway-suggestions/data-source-poller.php on line 47`